var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -288,6 +296,10 @@ def determine_variants(cls, exes, version_str):
                 compilers['cxx'] = exe
             elif 'clang' in exe:
                 compilers['c'] = exe
+            elif 'flang' in exe:
+                compilers['fc'] = exe
+            elif 'flang' in exe:
+                compilers['f77'] = exe
```

Comment:
```suggestion
            elif 'flang' in exe:
                compilers['fc'] = exe
                compilers['f77'] = exe
```
Also these should be combined in a single if-case